### PR TITLE
Add basic metal3 setup

### PR DESCRIPTION
**Pull request by lentzi90**
This is a PoC demonstrating a simple, even "minimal" setup of metal3. It is based on libvirt-backed BMHs combined with minikube also as a libvirt VM. This makes the network layout simple and easy to understand.

The readme contains all the information for how to use it. There are instructions for setting up the environment, how to create/delete BMHs, how to manage the BMHs directly (plain BMO, no CAPM3 or CAPI), and how to provision CAPI clusters in three different ways.

The setup aims to be non-invasive and not make changes on the host. When this is anyway needed, cleanup commands are provided.

Note: This looks way bigger than what it actually is. Most of the yaml is just copied straight from BMO with some slight adaptions to allow deploying it without using the `deploy.sh` script. Please disregard these files if you are not interested in details. The point of this PoC is mostly in the readme, not how exactly BMO and Ironic are deployed.

### CODE REVIEW

**Review by nsimons — COMMENTED** [review]
Few things I found when trying it out

File: basic-metal3-setup/README.md [hunk 1]
```diff
@@ -0,0 +1,383 @@
...
+      - path: /etc/modules-load.d/k8s.conf
+        owner: root:root
+        permissions: '0644'
+        content: |
+          br_netfilter
+    users:
+      - name: ubuntu
+        sudo: 'ALL=(ALL) NOPASSWD:ALL'
+        sshAuthorizedKeys:
+        - $(cat ~/.ssh/id_ed25519.pub)"
```
[the remark this answers is not in the document]
Comment (lentzi90) [reply basic-metal3-setup/README.md:238]: Good point! The default is still `id_rsa.pub` and it is probably not even created automatically. I should add some comment and/or steps to ensure it exists. (It is fine to use whatever public key you have on the host. This is just to get ssh access to the BMHs.)

File: basic-metal3-setup/README.md [hunk 2]
```diff
@@ -0,0 +1,383 @@
...
+./delete_bmh.sh host-3
+
+# Stop and remove image server
+docker stop image-server
+# Cleanup images
+rm --recursive images
+
+# Remove minikube ssh key from authorized_keys
+PUBLIC_KEY="$(minikube ssh -- cat .ssh/id_ed25519.pub)"
+sed "\#${PUBLIC_KEY}#d" ~/.ssh/authorized_keys
```
[the remark this answers is not in the document]
Comment (lentzi90) [reply basic-metal3-setup/README.md:379]: Good catch, thanks!

File: basic-metal3-setup/README.md [hunk 3]
```diff
@@ -0,0 +1,383 @@
...
+virsh net-destroy default
+echo "Add this to the network:"
+echo "<bootp file=http://${MINIKUBE_ETH1_IP}:6180/boot.ipxe/>"
+virsh net-edit default
+virsh net-start default
+# Start minikube again
+minikube start
+
+## VBMC with ssh connection to libvirt on the host
+# TODO: Ensure host has sshd running
```
[the remark this answers is not in the document]
Comment (lentzi90) [reply basic-metal3-setup/README.md:83]: Thanks! I added a prereq section at the top. Didn't go into all the details for how to setup and install every tool, but at least it should point in the right direction. :slightly_smiling_face:

**Review by tuminoid — APPROVED** [review]
Looks really good to me.

Won't be lgtm'ing it as we're all approvers and it'd merge it right away.

**Pull request merged**